Lib/test/test_typing.py

```diff
@@ -4656,6 +4672,15 @@ class Z(Generic[P]):
         self.assertEqual(G5.__parameters__, G6.__parameters__)
         self.assertEqual(G5, G6)
 
+        G7 = Z[int]
+        self.assertEqual(G7.__args__, ((int,),))
```

Comment:
It does not work for user generics because they can have more than one ParamSpec. For example:
```python
class X(Generic[P1, P2]):
    f: Callable[P1, int]
    g: Callable[P2, str]
G1 = X[[int, str], [bytes]]
G2 = X[int, [str, bytes]]
assert G1.__args__ == ((int, str), (bytes,))
assert G2.__args__ == ((int,) (str, bytes))
```
If flatten arguments, `__args__` of G1 and G2 would be the same (`(int, str, bytes)`). But they are clearly different types.

I'll add a special test for this.

Reply:
Right, thanks for pointing that out! I forgot there was ambiguity with multiple ParamSpec. Hopefully some future PEP makes this clearer.